### DISCUSSION TITLE
Some entities returning showing incorrect data while others correctly updating

### DISCUSSION THREAD

**Discussion by iceymanz** [ekutner/home-connect-hass#398] [Community Support] [answered]
I am opening a discussion rather than a bug because I'd like to know how to gather reliable data to troubleshoot this - is this the addon holding incorrect states or the API handing out garbage data. I do not want to waste ekutner's time.

My have a long running issue (currently on 1.1.7 but has existed for some time previously) where during a cycle, some sensor elements seem to get stuck on incorrect data. The device is a WAX32K41AU washing machine.

The attached image is a screenshot of the history of 4 different loads over a weekend. On the first load the top line 
binary_sensor.bosch_wax32k41au_xxxxxxx_laundrycare_washer_option_idos1_active is incorrect. This load was performed with iDos1 active.

Prior to wash 1 - idos1 is shown as off from 12:06 (arguably it's on as it's selected)
Wash 1 @ 12:35 - First wash cycle starts with idos1 active - idos state is not correctly updated. Note speedperfect is also not correct.
Wash 2 @ 13:25- the addon was restarted and it correctly detected idos1 on, idos2 off. 
Wash 3 @ 14:38 - Unclear why it went back to showing idos1 off when it should have been on. (Speedperfect not applicable to this cycle type)
Wash 4 @ 16:23 - idos1 off but idos 2 correctly correctly shows as active (I am using the dual detergent for wool wash because we don't use fabric softener in australia)

How do I gather the data to troubleshoot this as the debug button has been removed. Thank you.

![image](https://github.com/ekutner/home-connect-hass/assets/16264771/9bbbf818-1c6c-4619-868f-3e66031776e0)

Edit: Better image with sensor labels more visible. 
Ps - why care? Because this washing machine makes it too easy to accidentally run a cycle with incorrect or even no detergent.

**Comment by ekutner** [comment, marked as answer]
I appreciate the effort to avoid opening an issue but in this case it's not clear to me if this is a bug in the integration or HC. Please open an issue and attach a debug log with Log Level 7 (check the README) and specify the time of relevant events, like you did here.

**Comment by iceymanz** [comment]
Brief update, I will upload some debug logs and overview when I have a chance however key points:

* I captured one wash cycle 13/07 - with debug I can clearly see that at multiple points the bosch API was returning idos and speedperfect options that were not consistent with what was happening on the machine. 

* My guess would be that the machine isn't correctly updating the bosch service with the correct status information.

Co-incidentally - I received my first 'washing machine software update' notification - which did and it updated 'version' 3.4125-2.10.1.13125 (3.0) to (3.4125-2.15.1.13325 (3.0). 

The next 3 cycles, idos and speedperfect were correct. However I noticed that temperature was incorrect for one of two cycles (cold instead of 40c).

**Comment by ekutner** [comment]
So the current status is that this issue is a HC problem of reporting the wrong values?
